core/kernel/user_access.c

```diff
@@ -55,21 +80,78 @@ TEE_Result copy_from_user_private(void *kaddr, const void *uaddr, size_t len)
 
 	uaddr = memtag_strip_tag_const(uaddr);
 	res = check_access(flags, uaddr, len);
-	if (!res)
+	if (!res) {
+		enter_user_access();
 		memcpy(kaddr, uaddr, len);
+		exit_user_access();
+	}
 
 	return res;
 }
 
 TEE_Result copy_to_user_private(void *uaddr, const void *kaddr, size_t len)
 {
-	uint32_t flags = TEE_MEMORY_ACCESS_WRITE;
+	uint32_t flags = TEE_MEMORY_ACCESS_READ;
 	TEE_Result res = TEE_SUCCESS;
 
 	uaddr = memtag_strip_tag(uaddr);
 	res = check_access(flags, uaddr, len);
-	if (!res)
+	if (!res) {
+		enter_user_access();
 		memcpy(uaddr, kaddr, len);
+		exit_user_access();
+	}
+
+	return res;
+}
+
+size_t clear_user(void *uaddr, size_t n)
```

Comment:
why not returning a `TEE_Result`?

Reply:
Function type of `clear_user` was taken from other operating systems that have similar functions. updated the function to return `TEE_Result`